packages/radix-vue/src/ScrollArea/ScrollAreaViewport.vue

```diff
@@ -0,0 +1,61 @@
+<script setup lang="ts">
+import { inject, onMounted, ref } from "vue";
+import {
+  type ScrollAreaProvideValue,
+  SCROLL_AREA_INJECTION_KEY,
+} from "./ScrollAreaRoot.vue";
+
+const injectedValueFromRoot = inject<ScrollAreaProvideValue>(
+  SCROLL_AREA_INJECTION_KEY
+);
+
+const viewportElement = ref<HTMLElement>();
+const contentElement = ref<HTMLElement>();
+
+onMounted(() => {
+  injectedValueFromRoot?.onViewportChange(viewportElement.value!);
+  injectedValueFromRoot?.onContentChange(contentElement.value!);
+});
+</script>
+
+<template>
+  <div
+    ref="viewportElement"
+    data-radix-scroll-area-viewport=""
+    :style="{
+      /**
+       * We don't support `visible` because the intention is to have at least one scrollbar
+       * if this component is used and `visible` will behave like `auto` in that case
+       * https://developer.mozilla.org/en-US/docs/Web/CSS/overflowed#description
+       *
+       * We don't handle `auto` because the intention is for the native implementation
+       * to be hidden if using this component. We just want to ensure the node is scrollable
+       * so could have used either `scroll` or `auto` here. We picked `scroll` to prevent
+       * the browser from having to work out whether to render native scrollbars or not,
+       * we tell it to with the intention of hiding them in CSS.
+       */
+      overflowX: injectedValueFromRoot?.scrollbarXEnabled.value
+        ? 'scroll'
+        : 'hidden',
+      overflowY: injectedValueFromRoot?.scrollbarYEnabled.value
+        ? 'scroll'
+        : 'hidden',
+    }"
+  >
+    <div ref="contentElement" :style="{ minWidth: '100%', display: 'table' }">
```

Comment:
@zernonia I'm experiencing some layout issues with this component and it seems caused by this `display: 'table'`. What the reason behind using "table" here?

Reply:
We're not 100% sure regarding the reason for this, but I've tested out multiple case and seems like we don't really need it